internal/composer/openapi.v2.yml

```diff
@@ -822,6 +822,50 @@ components:
           description: |
             Naming package sets for a repository assigns it to a specific part
             (pipeline) of the build process.
+    CustomRepository:
+      type: object
+      required:
+        - id
+      description: |
+        Repository configuration for custom repositories.
+        At least one of the 'baseurl', 'mirrorlist', 'metalink' properties must
+        be specified. If more of them are specified, the order of precedence is
+        the same as listed above. Id is required.
+      properties:
+        id:
+          type: string
+        name:
+          type: string
+        filename:
+          type: string
+        baseurl:
+          type: array
+          example: [ 'https://cdn.redhat.com/content/dist/rhel8/8/x86_64/baseos/os/' ]
```

Comment:
I was in two minds about making this an array (since we changed it to an array in composer).
Maybe we could keep it as a string here and then convert it to an array internally so that this
and `payload_repositories` have a similar structure?